src/code/z_player_lib.c

```diff
@@ -2229,38 +2966,397 @@ struct_80128388_arg1 D_801C0C54[] = {
+    { 252, 238, 0 },   // PLAYER_BOTTLE_SEAHORSE         // PLAYER_IA_BOTTLE_SEAHORSE
+    { 131, 0, 174 },   // PLAYER_BOTTLE_MUSHROOM         // PLAYER_IA_BOTTLE_MUSHROOM
+    { 64, 64, 32 },    // PLAYER_BOTTLE_HYLIAN_LOACH     // PLAYER_IA_BOTTLE_HYLIAN_LOACH
+    { 0, 0, 255 },     // PLAYER_BOTTLE_BUG              // PLAYER_IA_BOTTLE_BUG
+    { 255, 0, 255 },   // PLAYER_BOTTLE_POE              // PLAYER_IA_BOTTLE_POE
+    { 255, 0, 255 },   // PLAYER_BOTTLE_BIG_POE          // PLAYER_IA_BOTTLE_BIG_POE
+    { 255, 0, 0 },     // PLAYER_BOTTLE_POTION_RED       // PLAYER_IA_BOTTLE_POTION_RED
+    { 0, 0, 255 },     // PLAYER_BOTTLE_POTION_BLUE      // PLAYER_IA_BOTTLE_POTION_BLUE
+    { 0, 200, 0 },     // PLAYER_BOTTLE_POTION_GREEN     // PLAYER_IA_BOTTLE_POTION_GREEN
+    { 255, 255, 255 }, // PLAYER_BOTTLE_MILK             // PLAYER_IA_BOTTLE_MILK
+    { 255, 255, 255 }, // PLAYER_BOTTLE_MILK_HALF        // PLAYER_IA_BOTTLE_MILK_HALF
+    { 255, 255, 255 }, // PLAYER_BOTTLE_CHATEAU          // PLAYER_IA_BOTTLE_CHATEAU
+    { 80, 80, 255 },   // PLAYER_BOTTLE_FAIRY            // PLAYER_IA_BOTTLE_FAIRY
 };
 
 Vec3f D_801C0CE8[PLAYER_FORM_MAX] = {
-    { 0.0f, 0.0f, 0.0f },    { 300.0f, 300.0f, -230.0f }, { 0.0f, 90.0f, -50.0f },
-    { 0.0f, 20.0f, -60.0f }, { 0.0f, 0.0f, 0.0f },
+    { 0.0f, 0.0f, 0.0f },        // PLAYER_FORM_FIERCE_DEITY
+    { 300.0f, 300.0f, -230.0f }, // PLAYER_FORM_GORON
+    { 0.0f, 90.0f, -50.0f },     // PLAYER_FORM_ZORA
+    { 0.0f, 20.0f, -60.0f },     // PLAYER_FORM_DEKU
+    { 0.0f, 0.0f, 0.0f },        // PLAYER_FORM_HUMAN
 };
-Vec3f D_801C0D24[PLAYER_FORM_MAX] = {
-    { 200.0f, 300.0f, 0.0f }, { 200.0f, 200.0f, 0.0f }, { 200.0f, 300.0f, 0.0f },
-    { 200.0f, 150.0f, 0.0f }, { 200.0f, 200.0f, 0.0f },
+
+void Player_DrawBunnyHood(PlayState* play) {
+    Mtx* temp_a1 = GRAPH_ALLOC(play->state.gfxCtx, 2 * sizeof(Mtx));
+    Vec3s sp2C;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gSPSegment(POLY_OPA_DISP++, 0x0B, temp_a1);
+
+    Matrix_Push();
+
+    sp2C.x = sBunnyEarKinematics.rot.y + 0x3E2;
+    sp2C.y = sBunnyEarKinematics.rot.z + 0xDBE;
+    sp2C.z = sBunnyEarKinematics.rot.x - 0x348A;
+    Matrix_SetTranslateRotateYXZ(97.0f, -1203.0f, -240.0f, &sp2C);
+
+    Matrix_ToMtx(temp_a1++);
+
+    sp2C.x = sBunnyEarKinematics.rot.y - 0x3E2;
+    sp2C.y = -sBunnyEarKinematics.rot.z - 0xDBE;
+    sp2C.z = sBunnyEarKinematics.rot.x - 0x348A;
+    Matrix_SetTranslateRotateYXZ(97.0f, -1203.0f, 240.0f, &sp2C);
+
+    Matrix_ToMtx(temp_a1);
+
+    Matrix_Pop();
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void func_80127B64(struct_801F58B0 arg0[], s32 count, Vec3f* arg2) {
+    s32 i;
+
+    for (i = 0; i < count; i++, arg0++) {
+        Math_Vec3f_Copy(&arg0->unk_00, arg2);
+        Math_Vec3f_Copy(&arg0->unk_0C, &gZeroVec3f);
+        arg0->unk_18 = 0;
+        arg0->unk_1A = 0;
+    }
+}
+
+void func_80127BE8(PlayState* play, Vec3f* arg1) {
```

Comment:
Maybe add a comment that this draws the great fairy mask particles when a stray fairy is in the room? Or just leave it as great fairy mask particles

Reply:
If that's all this is used for we might as well just name it.

Reply:
`Player_DrawStrayFairyParticles`?

Reply:
I'll just name this function